sdk/tables/Azure.Data.Tables/src/TableRestClient.cs

```diff
@@ -94,8 +90,10 @@ public async Task<Response<List<Response>>> SendBatchRequestAsync(HttpMessage me
                             if (match.Success && int.TryParse(match.Groups["index"].Value, out int failedEntityIndex))
                             {
                                 // create a new exception with the additional info populated.
-                                var appendedMessage = AppendEntityInfoToMessage(ex.Message, messageList[failedEntityIndex].Entity);
-                                throw new RequestFailedException(ex.Status, appendedMessage, ex.ErrorCode, ex.InnerException);
+                                var appendedMessage = AppendEntityInfoToMessage(ex.Message);
+                                var rfe = new RequestFailedException(ex.Status, appendedMessage, ex.ErrorCode, ex.InnerException);
```

Comment:
I'm assuming there are no serialization concerns that need to be taken into account here.

Reply:
Nope, the entity we stuff into the Data collection is never roundtripped.

Reply:
Hmm, well maybe there are. Framework tests are complaining when adding this to the exception.
 `  Error Message:    System.ArgumentException : Argument passed in is not serializable.`

I'll need to investigate further.